crates/common/src/io/macros.rs

```diff
@@ -0,0 +1,181 @@
+        }
+    }};
+}
+
+/// Prints a formatted error to stderr.
+///
+/// **Note**: will log regardless of the verbosity level.
+#[macro_export]
+macro_rules! sh_err {
+    ($($args:tt)*) => {
+        $crate::__sh_dispatch!(error $($args)*).expect("failed to write error")
+    };
+}
+
+/// Prints a formatted warning to stderr.
+///
+/// **Note**: if `verbosity` is set to `Quiet`, this is a no-op.
+#[macro_export]
+macro_rules! sh_warn {
+    ($($args:tt)*) => {
+        $crate::__sh_dispatch!(warn $($args)*).expect("failed to write warning")
+    };
+}
+
+/// Prints a raw formatted message to stdout.
+///
+/// **Note**: if `verbosity` is set to `Quiet`, this is a no-op.
+#[macro_export]
+macro_rules! sh_print {
+    ($($args:tt)*) => {
+        $crate::__sh_dispatch!(print_out $($args)*).expect("failed to write output")
+    };
+
+    ($shell:expr, $($args:tt)*) => {
+        $crate::__sh_dispatch!(print_out $shell, $($args)*).expect("failed to write output")
+    };
+}
+
+/// Prints a raw formatted message to stderr.
+///
+/// **Note**: if `verbosity` is set to `Quiet`, this is a no-op.
+#[macro_export]
+macro_rules! sh_eprint {
+    ($($args:tt)*) => {
+        $crate::__sh_dispatch!(print_err $($args)*).expect("failed to write error")
+    };
+
+    ($shell:expr, $($args:tt)*) => {
+        $crate::__sh_dispatch!(print_err $shell, $($args)*).expect("failed to write error")
+    };
+}
+
+/// Prints a raw formatted message to stdout, with a trailing newline.
+///
+/// **Note**: if `verbosity` is set to `Quiet`, this is a no-op.
+#[macro_export]
+macro_rules! sh_println {
+    () => {
+        $crate::sh_print!("\n")
+    };
+
+    ($fmt:literal $($args:tt)*) => {
+        $crate::sh_print!("{}\n", ::core::format_args!($fmt $($args)*))
+    };
+
+    ($shell:expr $(,)?) => {
+        $crate::sh_print!($shell, "\n").expect("failed to write newline")
+    };
+
+    ($shell:expr, $($args:tt)*) => {
+        $crate::sh_print!($shell, "{}\n", ::core::format_args!($($args)*)).expect("failed to write line")
+    };
+
+    ($($args:tt)*) => {
+        $crate::sh_print!("{}\n", ::core::format_args!($($args)*)).expect("failed to write line")
+    };
+}
+
+/// Prints a raw formatted message to stderr, with a trailing newline.
+///
+/// **Note**: if `verbosity` is set to `Quiet`, this is a no-op.
+#[macro_export]
+macro_rules! sh_eprintln {
+    () => {
+        $crate::sh_eprint!("\n")
+    };
+
+    ($fmt:literal $($args:tt)*) => {
+        $crate::sh_eprint!("{}\n", ::core::format_args!($fmt $($args)*))
+    };
+
+    ($shell:expr $(,)?) => {
+        $crate::sh_eprint!($shell, "\n")
+    };
+
+    ($shell:expr, $($args:tt)*) => {
+        $crate::sh_eprint!($shell, "{}\n", ::core::format_args!($($args)*))
+    };
+
+    ($($args:tt)*) => {
+        $crate::sh_eprint!("{}\n", ::core::format_args!($($args)*)).expect("failed to write line")
```

Comment:
The point of these macros is that we don't panic when we print stuff due to broken pipe, see issues which are supposed to be closed by this PR. Please revert these `expect` and keep the result, ignoring it is also fine with `let _ =` but try to propagate it where possible

Reply:
Done, good catch